### PR TITLE
[tune] catch SIGINT signal and trigger experiment checkpoint

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

With this PR, users can gracefully exit a tuning run anytime.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by richardliaw — APPROVED** [review]
no objections for now but would consider 
* a way to disable this
* some more documentation about this behavior change

do feel free to merge at will

**Comment by krfricke** [comment]
Great, thanks for the feedback. I added both, and a new paragraph in the user guide for resuming experiments.

File: doc/source/tune/user-guide.rst [hunk 1]
```diff
@@ -277,6 +278,60 @@ disable cross-node syncing:
...
+``~/ray_results/my_experiment``). If you forgot to pass a name in the first
+call, you can still pass the name when you resume the run. Please note that
+in this case it is likely that your experiment name has a date suffix, so if you
+ran ``tune.run(my_trainable)``, the ``name`` might look like something like this:
+``my_trainable_2021-01-29_10-16-44``.
+
+You can see which name you need to pass by taking a look at the results table
+of your original tuning run:
+
+.. code-block:: plain
```
Comment (krfricke) [inline doc/source/tune/user-guide.rst:324]:
```suggestion
.. code-block::
```

Comment (richardliaw) [reply doc/source/tune/user-guide.rst]: iirc you need to provide a language?

Comment (krfricke) [reply doc/source/tune/user-guide.rst]: I checked and didn't find anything for plain text. Let's see what the linter says

**Pull request merged**